test/tests/functional/pbs_daemon_service_user.py

```diff
@@ -0,0 +1,130 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2020 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import resource
+
+from tests.functional import *
+
+
+class TestDaemonServiceUser(TestFunctional):
+
+    """
+    Test suite to test running schedulers as a non-root user
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        self.orig_pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+
+    def tearDown(self):
+        self.du.set_pbs_config(self.server.shortname,
+                               confs=self.orig_pbs_conf,
+                               append=False)
```

Comment:
Why do you need to do this?  Won't revert to defaults revert pbs.conf?

Reply:
revert to defaults doesn't revert conf soon enough

Reply:
How can it not?  Your test doesn't start until after the revert happens.

Reply:
ptl fails to query fairshare data because it's trying to use the old pbs_daemon_Service_user value

Reply:
So revert pbs.conf sooner.  It should probably be the first thing reverted anyway

Reply:
ptl fails to delete all the jobs in teardown
In Server.cleanup_jobs, we status the server for all the schedulers. this causes us to create a new scheduler object in ptl, which calls pbsfs. this will fail because the conf file has PDSU set.

Reply:
You should fix this in setUp().  If you fix it this way, you are forcing any test which uses PDSU to revert it in tearDown().  This is PBS functionality, and it should be reverted properly.

Reply:
I don't see how we made it all the way to tearDown() without creating  all the schedulers.  They should have either been made in setUp() or in the test if it added any.  Why are we creating any in tearDown()?

Reply:
In teardown, the first thing we do is call `self.server.cleanup_jobs()`.
This calls `self.status(SCHED)`, which creates scheduler objects from the info returned by the status call.
In Scheduler's constructor, we query fairshare. The problem is in the test, we have set a custom PBS_DAEMON_SERVICE_USER in /etc/pbs.conf, and when we try to run pbsfs, pbsfs complains that the user we're running as (which is root by default) doesn't match PDSU.